ctapipe/calib/camera/calibrator.py

```diff
@@ -152,3 +181,36 @@ def __call__(self, event):
         for telid in tel.keys():
             self._calibrate_dl0(event, telid)
             self._calibrate_dl1(event, telid)
+
+
+def shift_waveforms(waveforms, time_shift_samples):
+    """
+    Shift the waveforms by the mean integer shift to mediate
+    time differences between pixels.
+    The remaining shift (mean + fractional part) is returned
+    to be applied later to the extracted peak time.
+    """
+    mean_shift = time_shift_samples.mean()
+    integer_shift = np.round(time_shift_samples - mean_shift).astype("int16")
+    remaining_shift = time_shift_samples - integer_shift
+    shifted_waveforms = _shift_waveforms_by_integer(waveforms, integer_shift)
+    return shifted_waveforms, remaining_shift
+
+
+@njit
+def _shift_waveforms_by_integer(waveforms, integer_shift):
```

Comment:
This could be implemented as a universal function instead (using @guvectorize)

Reply:
done